### DISCUSSION TITLE
Generating Input Files for Lake Reservoirs with Additional Reservoirs

### DISCUSSION THREAD

**Discussion by Zhanwei-Liu** [iiasa/CWatM#169] [Q&A] [answered]
I am attempting to simulate additional reservoirs beyond the existing 5-arcmin inputs. Could you suggest alternative methods for creating inputs for Lake Reservoirs using a shapefile of water surface areas? Additionally, do you have any available scripts that could assist with this process? Thank you!

**Comment by mikhailsmilovic** [comment, marked as answer]
Guten Tag Zhanwei-Liu,
When adding new lakes/reservoirs, one updates six raster files (ex. TIF and NetCDF).
These rasters must have the same cells associated with the lake/reservoir.

### Here is how I have done it, using GIS software (ex. QGIS, ArcGIS Pro, ArcMap)
If you have a shapefile with lakes/reservoirs, associate each shape with the following features:

- waterBodyID: integer>0, be sure that it doesn't overlap with other lakes/reservoirs
- waterBodyTyp: 1=lake, 2=reservoir, 3=lake until reservoir at construction year
- waterBodyDis: average m<sup>3</sup>/s discharge, used to determine release levels.
- waterBodyArea: area in km<sup>2</sup> (square kilometres)
- waterBodyVolRes: volume in Mm<sup>3</sup> (million m<sup>3</sup>)
- waterBodyYear: first year of operation

Once your shapefile is outfitted with the six features, convert it to a raster and mosaic it with the previous raster (worth looking for overlaps, which would be problematic). We would encourage at least one non-lake/reservoir cell between lakes/reservoirs. 
Here, one should "snap to raster" and choose the extent/projection/cell size of another CWatM input (such as the previous waterBodyID file) to ensure everything is aligned. 
We show rasterizing and aligning a shapefile here: https://youtu.be/AH4RA0G4rG4 

Then, export a raster file (TIF) for each feature and include the path to these files in the settings file as shown below.

waterBodyID = $(PathLakesRes)/lakesResID.tif
waterBodyTyp = $(PathLakesRes)/lakesResType.tif
waterBodyDis = $(PathLakesRes)/lakesResDis.tif
waterBodyArea = $(PathLakesRes)/lakesResArea.tif
waterBodyVolRes = $(PathLakesRes)/lakesResVolRes.tif
waterBodyYear = $(PathLakesRes)/lakesResYear.tif

**Reply by Zhanwei-Liu** [reply]
Thanks a lot for your response. Your response is very useful for me.
